Fixes #263 - Added fetching and rendering of data from the backend to the frontend

lucacataldo opened the pull request:
Added the fetching & rendering of posts in frontend. This includes fetching post & participant data using jQuery (currently dummy data in src/frontend/dummy_data),  and rendering it using a templated <article> element. I picked some posts from the current planet that included elements I though would bring up some edge cases and adjusted CSS accordingly so that when this gets hooked into the actual routes it'll be a bit less garbled and legible. Also, since pagination.js dialogue.js are no longer relevant or being used, I removed them.

cindyledev commented:
I also think everything in `main.css` can be a separate PR by itself as well

humphd commented on src/frontend/dummy_data/participants.json:
```diff
@@ -0,0 +1,84 @@
+  "participants": [
+    {
+      "name": "John Smith",
+      "url": "https:google.ca"
```
Let's use proper URLs throughout.

lucacataldo replied:
Good catch, thanks!

humphd commented on src/frontend/dummy_data/posts.json:
```diff
@@ -0,0 +1,36 @@
+{
+  "NOTE": "THIS IS TEMPORARY DUMMY DATA UNTIL DATA IS EXPOSED BY SERVER",
```
Remove this line

humphd commented on src/frontend/dummy_data/posts.json:
```diff
@@ -0,0 +1,36 @@
+      "title": "A Telescope Blog Post A Telescope Blog Post",
+      "author": "Tim Berners Lee",
+      "content": "<p>Hello friends,</p>\n\n\n\n<p></p>\n\n\n\n<p>October is right around the corner and that means a lot of different &#8220;fests&#8221; are here. The one that has myself really excited for is <a href=https://hacktoberfest.digitalocean.com/>Hacktober Fest</a>. The premise is that it promotes open-source projects and development by having developers create a minimum of four pull r ...
+      "updated": "October 15, 2019",
```
Can you make these proper JSON style date strings?

lucacataldo replied:
Good idea, I've now used the Date object methods to turn a properly formatted string while using the correct JS formatted Timestamp to create the Date object.

humphd commented on src/frontend/dummy_data/posts.json:
```diff
@@ -0,0 +1,36 @@
+      "published": "October 15, 2019",
+      "url": "https://neilong31.wordpress.com/2019/09/13/web-based-note-taker-forevernote/",
+      "site": "https://neilong31.wordpress.com/",
+      "guid": "null"
```
Should be something other than null.  Our guids look like `"tag:blogger.com,1999:blog-3586542501367863430.post-1706207209862339192"`, basically a URI of some type.

humphd commented on src/frontend/index.html:
```diff
@@ -54,216 +60,111 @@
+
+        nav.css({
+          transform: 'translateX(-200px)',
+          boxShadow: '0px 0px 100px var(--box-shadow-color)',
```
CSS in `.css` files please.  Let's use classes if possible.

lucacataldo replied:
Added classes, now using jQuery .addClass and .removeClass methods to swap states.

humphd commented on src/frontend/index.html:
```diff
@@ -54,216 +60,111 @@
+        close.click(function() {
+          nav.css({
+            transform: 'translateX(0px)',
+            boxShadow: '0px 0px 0px var(--box-shadow-color)',
```
Same thing here.

humphd commented on src/frontend/index.html:
```diff
@@ -54,216 +60,111 @@
 
-        let close = document.querySelector('#mobile-burger-close');
+    <script>
+      $(() => {
```
JavaScript in `.js` files please.

lucacataldo replied:
Moved to main.js, would appreciate any resources on how to keep code in the front end in modular files as I was struggling with that

humphd commented on src/frontend/style/main.css:
```diff
@@ -71,7 +71,7 @@ nav a:visited {
   width: 150px;
   background: var(--nav-btn-bgcolor);
-  font-size: 1.25em;
+  font-size: 1em;
```
Can these CSS changes get done in another PR?  They seem unrelated to this fix.

lucacataldo commented:
Thanks for the feedback @cindyledev @miggs125 @humphd its greatly appreciated! Will be putting up a new commit in the coming days addressing the issues brought up. 

In regards to having the Javascript in separate files, I had all the code separated into modular files initially (UI tools in one, fetching in another, etc.) but was getting a lot of eslint errors where it was picking up variables as unused between files because they weren't really linked anywhere other than the <script> tags in the HTML. Wasn't able to fix that so would it be alright to have everything in one (separate) .js file for now and maybe have someone file an issue / PR to make things more modular and correctly integrated or adjust eslint settings in the future?

lucacataldo commented:
So, I've got a commit waiting and ready to go with all the fixes specified including dropping the CSS changes. I previously added these to address UX / UI issues that arose when I added dummy data to better represent typical posts. The following is a screenshot of a mobile view with the proposed changes:

![image](https://user-images.githubusercontent.com/35984346/69779628-9e6d5580-1176-11ea-8b90-3b365c369676.png)

As you can see this will greatly impact the way that posts are rendered. 


@humphd @miggs125 @cindyledev 
I'll leave it up to you guys as to whether or not to include them or not since I can personally see valid arguments for both including and not including them. 🤷‍♂ 

I think everything else that was mentioned has been fixed and will put up the PR when I hear back regarding the CSS changes. Thanks again for taking the time to look things over!

humphd commented:
@lucacataldo the discussions of splitting this up aren't aimed at having you throw changes away; rather, we're trying to improve the reviewability of the work, and isolate changes to their own PRs.  Doing the fixes to CSS is important, and we need that.  But the question is: do we need it in the same PR as one which adds data fetching?

If it's not possible for you to separate these, then please go ahead and do it all at once.  But in future, I want all of us to try and think about how to keep our changes simple, focused on doing only 1 thing, and making it easier for them to get reviewed, tested, and merged.

lucacataldo commented:
@humphd totally understandable and it was a mistake on my part to go down the rabbit hole so to speak. Will definitely keep this in mind in the future. As for now, I am going to give it a shot to minimize the amount of CSS added in this PR and file a new issue for the other smaller fixes.

lucacataldo commented:
@cindyledev @humphd @miggs125 changes have been pushed, have a look and let me know if there's anything that's still outstanding!

humphd commented on package.json:
```diff
@@ -23,8 +23,11 @@
-    "jest-watch":"cross-env MOCK_REDIS=1 jest --watch",
-    "start": "node src",
+    "jest-watch": "cross-env MOCK_REDIS=1 jest --watch",
+    "start": "node src/backend",
```
None of these files should be changed in your branch as far as I can tell.  We've had people overwrite these already in the past few days, I'm not eager to repeat it.  If I'm wrong and you do need these to be updated, let me know why.

humphd commented:
@lucacataldo see my work in #426, which takes what you have here, and connects it into the new backend we just merged.  You could use that to update your code in this PR.

humphd commented:
Closing and continuing in my other PR with new changes added.
